File: spoj/00-classical/AP2.rb
gets.to_i.times do
    third, third_last, sum = gets.split(' ').map(&:to_i)

    n = (2*sum)/(third+third_last)
    r = (third_last - third)/(n-5)

    puts n

    first = third-2*r
    for i in 0..n-1 do
        print first + i*r, " "
    end
end

File: spoj/AP2.py
import sys

for _ in range(input()):
    third, third_last, total = map(int, raw_input().split())
    
    n = (2*total)/(third+third_last)
    r = (third_last - third)/(n - 5)
    
    print n
    out = ""
    first = third - 2*r
    for i in range(0, n): out += str(first+i*r) + " "
        
    print out